analysis/ic50_analysis.py

```diff
@@ -0,0 +1,62 @@
+import json
+import sys
+from typing import Any
+
+from ic50 import *
+
+
+def main(argv: Any) -> None:
+    data = json.loads(argv[0])
+
+    response = []
+
+    for experiment in data:
+        response_value = {"experiment": experiment["experiment"], "status": "DONE"}
+        measurements = np.array(experiment['data'])
+
+        try:
+            find_outliers = experiment["find_outliers"]
+            if type(find_outliers) == str:
+                find_outliers = find_outliers.lower()
+                find_outliers = int(find_outliers == 'true')
+        except KeyError:
+            find_outliers = True
+
+        if measurements.shape[1] != 2:
+            response_value["status"] = "ERROR: Wrong experiment format."
+            response.append(response_value)
+            continue
+
+        if np.unique(measurements[:, 0]).shape[0] < 4:
+            response_value["status"] = "ERROR: Too little data."
+            response.append(response_value)
+            continue
+
+        try:
+            params = calculate_sigmoid(measurements)
+            response_value["params"] = params
+        except (ArithmeticError, RuntimeError):
+            response_value["status"] = "ERROR: Can not fit sigmoid."
```

Comment:
I think we should somehow show error message. So, my idea was to show full error on backend logs. In this case we want to have full error message in `stderr`. But I don't know if it is good or bad solution. cc @gromakovsky

Reply:
I think printing detailed error messages to `stderr` is a good idea. They won't show up in UI, but will show up in backend logs. So `response_value["status"]` is for short error messages that can be shown to user (currently we will show them as is, later we may parse them from Haskell and show differently if we want) and `stderr` is for logs and you can print anything there as long as you don't print _way too much_.

Reply:
@sashasashasasha151 Do I simply print it with `print()`? Or or go for: `print(*args, file=sys.stderr, **kwargs)`? The second option will just format it as an error. Don't know how backend interprets the output.

Reply:
Backend reads JSON output from `stdout` and parses it in JSON. If you print something else into `stdout`, it will break parsing. And `stderr` output is catched by the backend and printed into the backend logs. So I think `stderr_print(exc)` is good.